kernel/src/lib.rs

```diff
@@ -415,3 +418,70 @@ pub trait Engine: AsAny {
     /// Get the connector provided [`ParquetHandler`].
     fn get_parquet_handler(&self) -> Arc<dyn ParquetHandler>;
 }
+
+#[cfg(doctest)]
+mod doc_tests {
```

Comment:
Another question: For cases where compilation and execution succeed, should we use normal tests instead of doc tests? We only _need_ doc tests for cases that expect compilation failure, right?

Reply:
yes, exactly, that's why I only wanted to test for failure. I kinda shot myself in the foot by adding successful compiles :)

Reply:
How about this: I am going to leave only one successful macro compile doctest just to assure the methodology is working. Successful tests can be implemented as normal integration tests. What say you @scovich?

Reply:
Sounds great, thanks. If you decide not to tackle those integration tests, can you file a tracking ticket for the TODO?